nexus/internal/filetransfer/tasks.go

```diff
@@ -32,3 +32,21 @@ type UploadTask struct {
 	// Callback to execute after completion (success or failure).
 	CompletionCallback func(*UploadTask)
 }
+
+// DownloadTask is a task to download a file
```

Comment:
why do you need this one if it is exactly the same as download task?

Reply:
Initially I thought things might diverge but for now, i agree its better to keep things consolidated.